server/controllers/userAuth.js

```diff
@@ -301,9 +320,62 @@ const updatePassword = (req, resp) => {
     });
 };
 
+const resendRecoveryLink = (req, resp, next) => {
+  const { token } = req.params;
+  const sanitizedToken = sanitize(token);
+  const newToken = crypto.randomBytes(32).toString('hex');
+  const newExpirationDate = new Date().getTime() + 3600000;
+
+  User.findOneAndUpdate(
+    {
+      resetToken: sanitizedToken
+    },
+    {
+      resetToken: newToken,
+      resetTokenExpirationDate: newExpirationDate
+    }
+  )
+    .lean()
+    .exec()
+    .then(user => {
+      if (!user) {
+        throw new Error('authorization.actions.reset');
+      }
+      const { isActive } = user;
+
+      if (!isActive) {
```

Comment:
You can disable resetting password for inactive accounts in query":
```
  User.findOneAndUpdate(
    {
      resetToken: sanitizedToken,
      isActive: true
    },
(...)
```
Of course, we won't be able to return a specific message for this case as you did in line 348, but we try to avoid setting messages Id on the backend and in I'm wondering if this is really necessary for this situation...

Reply:
I did it on purpose. Otherwise, user will have no idea why he can not reset the password

Reply:
I know you did it on purpose. But user with inactive account can't reset password: in resetPassword controller, we check isActive field and if it is false we throw an error and don't send an email with the link. So I think such a user won't be able to access linkExpired page with a proper token in URL resending link will fail. So, in my opinion, sending this additional message is unnecessary here. But I maybe I didn't understand this logic correctly...

Reply:
@marekrozmus  what do you think?
Should we send here specific message?
Or just add isActive to query and throw a basic error when user account is not activated?

Reply:
if account is not activated or account does not exist send some common error like: "account does not exist or not activated" - user should know if he/she activated/created account.

Reply:
Done